src/DI/OrmExtension.php

```diff
@@ -129,26 +130,42 @@ public function loadEntityManagerConfiguration(): void
 			throw new InvalidStateException(sprintf('EntityManagerDecorator class "%s" not found', $entityManagerDecoratorClass));
 		}
 
-		// Entity Manager
-		$original = $builder->addDefinition($this->prefix('entityManager'))
-			->setType(DoctrineEntityManager::class)
-			->setFactory(DoctrineEntityManager::class . '::create', [
-				$builder->getDefinitionByType(Connection::class), // Nettrine/DBAL
-				$this->prefix('@configuration'),
-			])
-			->setAutowired(false);
-
-		// Entity Manager Decorator
-		$builder->addDefinition($this->prefix('entityManagerDecorator'))
-			->setFactory($entityManagerDecoratorClass, [$original]);
-
-		// ManagerRegistry
-		$builder->addDefinition($this->prefix('managerRegistry'))
-			->setType(ManagerRegistry::class)
-			->setArguments([
-				$builder->getDefinitionByType(Connection::class),
-				$this->prefix('@entityManagerDecorator'),
-			]);
+		foreach ($builder->findByType(\Doctrine\DBAL\Connection::class) as $definitionName => $serviceDefinition) {
+			$match = Strings::match($definitionName, '~([a-zA-Z]+\.([a-zA-Z]+))\.connection~');
```

Comment:
So it would work only if connection is defined with name matching this regex?

Reply:
@mabar any another idea, how to get connection name from DBAL?

Reply:
`$definitionName` is all you need.